### DISCUSSION TITLE
use modal in form action?

### DISCUSSION THREAD

**Discussion by stimw** [edmundhung/conform#458] [Q&A] [answered]
Hi there,

If I want to submit through the form's action and display a confirmation modal before submitting, do I have to hide the button and then use useRef like this: 

```js
<button type="submit" style={{ display: 'none' }} id="hiddenSubmit"></button>
```
and
```js
formRef.current.submit();
```

or there's some easier approaches?

**Comment by edmundhung** [comment, marked as answer]
You can put a submit button in the model and use the [form attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#form) to associate it with the form, e.g.

```tsx
// In the component
const [form,fields] = useForm();

// In the dialog
<button form={form.id}>Confirm</button>
```

If it is hard to pass the id down, you can always hard code it as long as it is unique.
```tsx
// In the component
const [form,fields] = useForm({
  id: 'example',
});

// In the dialog
<button form="example">Confirm</button>
```

**Reply by stimw** [reply]
looks good to me, thank you
